cli-toolchain/graphql.md

```diff
@@ -2379,6 +2379,157 @@ Here is a complete list of searchable operations per GraphQL type supported as o
+          translateText
+        }
+```
+
+#### Usage
+
+
+Given the following schema a query operation is defined which will do the following with the provided image.
+
+- Identify text from the image
+- Translate the text from that image
+- Synthesize speech from the translated text.
+
+```graphql
+type Query {
+  speakTranslatedImageText: String @predictions(actions: [
+    identifyText
+    translateText
+    convertTextToSpeech
+  ])
+}
+```
+
+An example of that query will look like:
+
+```graphql
+query SpeakTranslatedImageText($input: SpeakTranslatedImageTextInput!) {
+  speakTranslatedImageText(input: {
+    identifyText: {
+      key: "myimage.jpg"
+    }
+    translateText: {
+      sourceLanguage: "en"
+      targetLanguage: "es"
+    }
+    convertTextToSpeech: {
+      voiceID: "Conchita"
+    }
+  })
+}
+```
+
+A code example of this using the JS Library:
+```js
+import React, { useState } from 'react';
+import './App.css';
+import API, { graphqlOperation } from '@aws-amplify/api';
+import Amplify, { Storage } from 'aws-amplify';
+import awsconfig from './aws-exports';
+import { speakTranslatedImageText,speakLabels } from './graphql/queries';
+
+/* Configure Exports */
+Amplify.configure(awsconfig);
+
+function SpeakTranslatedImage() {
+  const [ src, setSrc ] = useState("");
+  const [ img, setImg ] = useState("");
+  
+  function putS3Image(event) {
+    const file = event.target.files[0];
+    Storage.put(file.name, file)
+    .then (async (result) => {
+      setSrc(await speakTranslatedImageTextOP(result.key))
+      setImg(await Storage.get(result.key));
+    })
+    .catch(err => console.log(err));
+  }
+
+
+  return (
+    <div className="Text">
+      <div>
+        <h3>Upload Image</h3>
+        <input
+              type = "file" accept='image/jpeg'
+              onChange = {(event) => {
+                putS3Image(event)
+              }}
+          />
+        <br />
+        { img && <img src = {img}></img>}
+        { src && <div> <audio id="audioPlayback" controls>
+        <source id="audioSource" type="audio/mp3" src = {src} />
+      </audio> </div> }
+      </div>
+    </div>
+  );
+}
+
+async function speakTranslatedImageTextOP(key) {
+  const inputObj = { translateText: { sourceLanguage: "en", targetLanguage: "es" }, identifyText: { key }, convertTextToSpeech: { voiceID: "Conchita" } };
+  const response = await API.graphql(graphqlOperation(speakTranslatedImageText, { input: inputObj }));
+  return response.data.speakTranslatedImageText;
+}
+function App() {
+  return (
+    <div className="App">
+        <h1>Speak Translated Image</h1>
+        < SpeakTranslatedImage />
+        <h1>Speak Labels</h1>
+        < SpeakLabels />
+    </div>
+  );
+}
+export default App;
+```
+
+#### What happens in the backend
+
+From the provided schema above the action list `[ identifyText translateText convertTextToSpeech ]`. The following resources are created to communicate with Amazon Rekognition, Translate and Polly.
```

Comment:
The example schema will communicate with Amzn Reko.... (actions: `[ identifyText translateText convertTextToSpeech ]`) to generate the following resources: 
- IAM Policy for each service (e.g. Amazon Rekognition `detectText` policy)
- An AppSync VTL function for each action
- A data source for each action